test/integration/controlling_test.exs

```diff
@@ -241,6 +241,30 @@ defmodule Oban.Integration.ControllingTest do
     :ok = stop_supervised(Oban)
   end
 
+  test "retrying jobs from multiple states" do
+    start_supervised_oban!(queues: [alpha: 10])
+
+    job_a = insert!(%{ref: 1}, state: "discarded", attempt: 20)
+    job_b = insert!(%{ref: 2}, state: "completed")
+    job_c = insert!(%{ref: 3}, schedule_in: 10)
+    job_d = insert!(%{ref: 4}, schedule_in: 10, state: "retryable")
+    job_e = insert!(%{ref: 5}, state: "executing", attempt: 1, max_attempts: 1)
+
+    assert :ok = Oban.retry_job(job_a.id)
+    assert :ok = Oban.retry_job(job_b.id)
+    assert :ok = Oban.retry_job(job_c.id)
+    assert :ok = Oban.retry_job(job_d.id)
+    assert :ok = Oban.retry_job(job_e.id)
+
+    assert %Job{state: "available", max_attempts: 21} = Repo.reload(job_a)
+    assert %Job{state: "available", max_attempts: 20} = Repo.reload(job_b)
```

Comment:
This relies on the somewhat arbitrary 20 retry limit. Let's set `max_attempts` for all of the various jobs instead to be explicit about it.